polkadot/zombienet-sdk-tests/tests/disabling/slashing.rs

```diff
@@ -0,0 +1,144 @@
+				.with_node(|node| node.with_name("honest-validator-1"))
+				.with_node(|node| node.with_name("honest-flaky-validator-0"))
+				.with_node(|node| {
+					node.with_name("malicious-backer")
+						.with_command("malus")
+						.with_subcommand("suggest-garbage-candidate")
+						.with_args(vec![
+							"--no-hardware-benchmarks".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace,parachain=debug".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(1337)
+				.with_default_command("polkadot-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-1337"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2025)
+				.with_default_command("polkadot-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-2025"))
+		})
+		.build()
+		.map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	let network = spawn_fn(config).await?;
+
+	let malus = network.get_node("malicious-backer")?;
+	malus.pause().await?;
+
+	let honest = network.get_node("honest-validator-0")?;
+	let relay_client: OnlineClient<PolkadotConfig> = honest.wait_client().await?;
+
+	// Wait for some para blocks being produced
+	assert_para_throughput(&relay_client, 2, [(ParaId::from(1337), 1..5)].into_iter().collect())
+		.await?;
+
+	// Let's initiate a dispute
+	malus.resume().await?;
+
+	let timeout_secs: u64 = 360;
+	let disputes_total_metric = "polkadot_parachain_candidate_disputes_total";
+	wait_for_metric(&honest, disputes_total_metric, 1).await;
+
+	// Pause first flaky node, so they stop concluding
+	let flaky_0 = network.get_node("honest-flaky-validator-0")?;
+	flaky_0.pause().await?;
+
+	// Wait for a dispute that will not conclude
+	let disputes = wait_for_metric(&honest, disputes_total_metric, 2).await;
+	log::info!("Dispute initiated, now waiting for a session change");
+
+	wait_for_session_change(&relay_client).await?;
+
+	// Now resume flaky validator
+	log::info!("Resuming the flaky node - disputes should conclude");
+	flaky_0.resume().await?;
+	// We don't need malus anymore
+	malus.pause().await?;
+
+	wait_for_metric(&honest, "polkadot_parachain_candidate_dispute_concluded{validity=\"invalid\"}", disputes).await;
+	log::info!("A dispute has concluded");
+
+	honest
+		.wait_log_line_count_with_timeout(
+			"*Successfully reported pending slash*",
+			true,
+			1,
+			timeout_secs,
+		)
+		.await?;
+
+	assert_blocks_are_being_finalized(&relay_client).await?;
+
+	log::info!("Test finished successfully");
+
+	Ok(())
+}
+
+pub async fn wait_for_metric(node: &NetworkNode, metric: &str, value: u64) -> u64 {
```

Comment:
Hi @ordian can we just use the `wait_metric` fn provided by the sdk? Something like this will work
```rs
let disputes = wait_for_metric(&honest, disputes_total_metric, |x| x >= 2 ).await;
```
Thx!

Reply:
yes, i used it initially - see the first commit, but for some reason it didn't work..

Reply:
Oks, I can take a look and make it works. Thx!

Reply:
in general metrics fetching is somehow unstable

Reply:
Ahh, I would double check this. Thanks!